### PR TITLE
[bitnami/mongodb] Add medium option for non-persisted.

### LINKED ISSUE #7382: [bitnami/mongodb] Option for using emptyDir with medium: memory
**Which chart**:
mongodb

**Is your feature request related to a problem? Please describe.**
I am using mongodb as a cache running in memory only.

**Describe the solution you'd like**
I would like to set the emptyDir which is used in case of `persistence: false` to `medium: memory`.

**Describe alternatives you've considered**
Well, using something else than mongo. :)

**Additional context**

### PR DESCRIPTION
**Pull request by MikaelElkiaer**
**Description of the change**
Adds `persistence.medium` and `hidden.persistence.medium` options.

**Benefits**
With the added medium option, it is now possible to set medium to `Memory`, as opposed to defaulting to local fs.

**Possible drawbacks**
None come to mind, as this is completely optional and will keep the default if not set.

**Applicable issues**

  - closes #7382 

**Additional information**

**Checklist** 

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/mongodb/Chart.yaml [hunk 1]
```diff
@@ -26,4 +26,4 @@ name: mongodb
 sources:
   - https://github.com/bitnami/bitnami-docker-mongodb
   - https://mongodb.org
-version: 10.29.4
+version: 10.29.5
```
Comment (miguelaeh) [inline bitnami/mongodb/Chart.yaml:29]: Adding new features should be a minor bump
```suggestion
version: 10.30.0
```

File: bitnami/mongodb/templates/hidden/statefulset.yaml [hunk 2]
```diff
@@ -511,7 +511,13 @@ spec:
         {{- end }}
   {{- if not .Values.hidden.persistence.enabled }}
         - name: datadir
+          {{- if .Values.hidden.persistence.medium }}
+          emptyDir: {
+            medium: {{ .Values.hidden.persistence.medium | quote }}
+          }
```
Comment (miguelaeh) [inline bitnami/mongodb/templates/hidden/statefulset.yaml:517]:
```suggestion
          emptyDir:
            medium: {{ .Values.hidden.persistence.medium | quote }}
```

File: bitnami/mongodb/templates/replicaset/statefulset.yaml [hunk 3]
```diff
@@ -545,7 +545,13 @@ spec:
         {{- end }}
   {{- if not .Values.persistence.enabled }}
         - name: datadir
+          {{- if .Values.persistence.medium }}
+          emptyDir: {
+            medium: {{ .Values.persistence.medium | quote }}
+          }
```
Comment (miguelaeh) [inline bitnami/mongodb/templates/replicaset/statefulset.yaml:551]:
```suggestion
          emptyDir:
            medium: {{ .Values.persistence.medium | quote }}
```

File: bitnami/mongodb/templates/standalone/dep-sts.yaml [hunk 4]
```diff
@@ -480,7 +480,13 @@ spec:
         {{- end }}
   {{- if not .Values.persistence.enabled }}
         - name: datadir
+          {{- if .Values.persistence.medium }}
+          emptyDir: {
+            medium: {{ .Values.persistence.medium | quote }}
+          }
```
Comment (miguelaeh) [inline bitnami/mongodb/templates/standalone/dep-sts.yaml:486]:
```suggestion
          emptyDir:
            medium: {{ .Values.persistence.medium | quote }}
```

**Review by miguelaeh — CHANGES_REQUESTED** [review]
Thank you for the PR! Please check the comments

**Comment by MikaelElkiaer** [comment]
> Thank you for the PR! Please check the comments

Thanks for the comments.
Obviously a minor, and not a patch, d'oh.

In regards to the yaml style, I actually adopted it from the redis chart.

**Review by miguelaeh — APPROVED** [review]
LGTM!

**Pull request merged**